### PR TITLE
Add Free JS Book check_urls=free-programming-bookslangs.md

### PR DESCRIPTION
**Pull request by VasekS**
## What does this PR do?
Add Free JS Book

### Description
JavaScript

### Why is this valuable (or not)?
Lots of good material

### How do we know it's really free?
Free PDF

### For book lists, is it a book? For course lists, is it a course? etc.
PDF Book

## Checklist:
- [Y ] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ Y] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [ Y] Include author(s) and platform where appropriate.
- [Y ] Put lists in alphabetical order, correct spacing.
- [Y ] Add needed indications (PDF, access notes, under construction).
- [ Y] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1081,6 +1081,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [JavaScript ES6 and beyond](https://github.com/AlbertoMontalesi/JavaScript-es6-and-beyond-ebook) - Alberto Montalesi (PDF, epub)
 * [JavaScript For Cats](http://jsforcats.com) - Maxwell Ogden (HTML)
 * [JavaScript for Impatient Programmers (ES2020 edition)](https://exploringjs.com/impatient-js/toc.html) - Dr. Axel Rauschmayer (HTML)
+* [JavaScript for Impatient Programmers (ES2023 edition)](https://exploringjs.com/impatient-js/downloads/impatient-js-preview-book.pdf) - Dr. Axel Rauschmayer (PDF)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1084]: We do not include honorifics like  "Dr."

Comment (VasekS) [reply books/free-programming-books-langs.md]: Hey @eshellman this should be updated now!

Comment (davorpa) [reply books/free-programming-books-langs.md]: PDF is a preview 50%. Please... use landing page

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -1081,6 +1081,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [JavaScript ES6 and beyond](https://github.com/AlbertoMontalesi/JavaScript-es6-and-beyond-ebook) - Alberto Montalesi (PDF, epub)
 * [JavaScript For Cats](http://jsforcats.com) - Maxwell Ogden (HTML)
 * [JavaScript for Impatient Programmers (ES2020 edition)](https://exploringjs.com/impatient-js/toc.html) - Dr. Axel Rauschmayer (HTML)
+* [JavaScript for Impatient Programmers (ES2023 edition)](https://exploringjs.com/impatient-js/downloads/impatient-js-preview-book.pdf) - Axel Rauschmayer (PDF)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:1084]: PDF is a preview 50%. Please... use landing page which links all versions, including HTML having full content

https://exploringjs.com/impatient-js/

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**